From f29ae58ddc93d86504c7e646ec5e73befb36594c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 7 Jun 2023 17:16:50 +0200
Subject: [PATCH] Respect CC, CFLAGS, and LDFLAGS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The distributions use these environment variables to compile their
software uniformly. This patch modifies the ccc scripts to respect
them.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 bfp/bfp_LUSOL/ccc | 20 +++++++++-----------
 demo/ccc          |  7 +++----
 lp_solve/ccc      | 13 ++++++-------
 lpsolve55/ccc     | 14 ++++++--------
 4 files changed, 24 insertions(+), 30 deletions(-)

diff --git a/bfp/bfp_LUSOL/ccc b/bfp/bfp_LUSOL/ccc
index 754738b..3ec9596 100644
--- a/bfp/bfp_LUSOL/ccc
+++ b/bfp/bfp_LUSOL/ccc
@@ -1,37 +1,35 @@
 :
 src='../../shared/commonlib.c ../../colamd/colamd.c lp_LUSOL.c ../../lp_utils.c ../../shared/myblas.c LUSOL/lusol.c'
 
-c=cc
+c=${CC:-cc}
+CFLAGS=${CFLAGS:--O3}
 
 #determine platform (32/64 bit)
 >/tmp/platform.c
 echo '#include <stdlib.h>'>>/tmp/platform.c
 echo '#include <stdio.h>'>>/tmp/platform.c
 echo 'main(){printf("ux%d", (int) (sizeof(void *)*8));}'>>/tmp/platform.c
-$c /tmp/platform.c -o /tmp/platform
+$c $CFLAGS /tmp/platform.c $LDFLAGS -o /tmp/platform
 PLATFORM=`/tmp/platform`
 rm /tmp/platform /tmp/platform.c >/dev/null 2>&1
 
 mkdir bin bin/$PLATFORM >/dev/null 2>&1
 
-opts='-O3'
-#opts=''
-
 if [ "$PLATFORM" = "RS6000" ]
-then opts="$opts"
+then
      so="-bM:SRE -bE:../bfp_lpsolve.exp"
      #so="$so -e _nostart"
      so="$so -bnoentry"
 elif [ "$PLATFORM" = "SCO_UNIX" ]
 then
-     opts='-O0'
      so="-G -dy"
-     opts="$opts -DNOLONGLONG $so"
-else opts="$opts -fpic"
+     CFLAGS="${CFLAGS} -O0 -DNOLONGLONG $so"
+else
+     CFLAGS="${CFLAGS} -fpic"
      so="-shared -Wl,-Bsymbolic -Wl,-soname,libbfp_LUSOL.so"
 fi
 
-$c -s -c $opts -I.. -I../.. -I../../colamd -I../../shared -ILUSOL -I. -DRoleIsExternalInvEngine -DINVERSE_ACTIVE=INVERSE_LUSOL $src
-$c $so -o bin/$PLATFORM/libbfp_LUSOL.so `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` -lc -lm -ldl
+$c $CFLAGS -c -I.. -I../.. -I../../colamd -I../../shared -ILUSOL -I. -DRoleIsExternalInvEngine -DINVERSE_ACTIVE=INVERSE_LUSOL $src
+$c $LDFLAGS $so -o bin/$PLATFORM/libbfp_LUSOL.so `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` -lc -lm -ldl
 
 rm *.o >/dev/null
diff --git a/demo/ccc b/demo/ccc
index 38713a1..154b898 100644
--- a/demo/ccc
+++ b/demo/ccc
@@ -1,5 +1,6 @@
 src='../lp_MDO.c ../shared/commonlib.c ../colamd/colamd.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
-c=cc
+c=${CC:-cc}
+CFLAGS=${CFLAGS:--O3}
 
 math=-lm
 
@@ -10,6 +11,4 @@ then def='-dy -K PIC -DNOLONGLONG'
 else dl=-ldl
 fi
 
-opts='-O3'
-
-$c -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine demo.c $src -o demo $math $dl
+$c $CFLAGS -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine demo.c $src $LDFLAGS -o demo $math $dl
diff --git a/lp_solve/ccc b/lp_solve/ccc
index bd5a938..fbb67d9 100644
--- a/lp_solve/ccc
+++ b/lp_solve/ccc
@@ -1,6 +1,7 @@
 :
 src='../lp_MDO.c ../shared/commonlib.c ../colamd/colamd.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../fortify.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c lp_solve.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
-c=cc
+c=${CC:-cc}
+CFLAGS=${CFLAGS:--O3}
 
 MYTMP=`mktemp -d "${TMPDIR:-/tmp}"/lp_solve_XXXXXX`
 
@@ -9,7 +10,7 @@ MYTMP=`mktemp -d "${TMPDIR:-/tmp}"/lp_solve_XXXXXX`
 echo '#include <stdlib.h>'>>"$MYTMP"/platform.c
 echo '#include <stdio.h>'>>"$MYTMP"/platform.c
 echo 'main(){printf("ux%d", (int) (sizeof(void *)*8));}'>>"$MYTMP"/platform.c
-$c "$MYTMP"/platform.c -o "$MYTMP"/platform
+$c $CFLAGS "$MYTMP"/platform.c $LDFLAGS -o "$MYTMP"/platform
 PLATFORM=`"$MYTMP"/platform`
 rm "$MYTMP"/platform "$MYTMP"/platform.c >/dev/null 2>&1
 
@@ -23,23 +24,21 @@ echo '#include <stdio.h>'>>"$MYTMP"/isnan.c
 echo '#include <stdlib.h>'>>"$MYTMP"/isnan.c
 echo '#include <math.h>'>>"$MYTMP"/isnan.c
 echo 'main(){isnan(0.0);return 0;}'>>"$MYTMP"/isnan.c
-$c "$MYTMP"/isnan.c -o "$MYTMP"/isnan $math >/dev/null 2>&1
+$c $CFLAGS "$MYTMP"/isnan.c $LDFLAGS -o "$MYTMP"/isnan $math >/dev/null 2>&1
 if [ $? = 0 ]
 then NOISNAN=
 else NOISNAN=-DNOISNAN
 fi
 rm "$MYTMP"/isnan.c "$MYTMP"/isnan >/dev/null 2>&1
 
-opts='-O3'
-
 rmdir "$MYTMP"
 
 def=
 if [ "$PLATFORM" = "SCO_UNIX" ]
-then opts='-O0'
+then CFLAGS="${CFLAGS} -O0"
      def='-dy -K PIC -DLLONG=long'
      dl=-ldl
 else dl=-ldl
 fi
 
-$c -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src -o bin/$PLATFORM/lp_solve $math $dl
+$c $CFLAGS -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src $LDFLAGS -o bin/$PLATFORM/lp_solve $math $dl
diff --git a/lpsolve55/ccc b/lpsolve55/ccc
index 999f5f6..49cd7dc 100644
--- a/lpsolve55/ccc
+++ b/lpsolve55/ccc
@@ -1,6 +1,6 @@
 :
 src='../lp_MDO.c ../shared/commonlib.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../fortify.c ../colamd/colamd.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
-c=cc
+c=${CC:-cc}
 
 MYTMP=`mktemp -d "${TMPDIR:-/tmp}"/lp_solve_XXXXXX`
 
@@ -9,7 +9,7 @@ MYTMP=`mktemp -d "${TMPDIR:-/tmp}"/lp_solve_XXXXXX`
 echo '#include <stdlib.h>'>>"$MYTMP"/platform.c
 echo '#include <stdio.h>'>>"$MYTMP"/platform.c
 echo 'main(){printf("ux%d", (int) (sizeof(void *)*8));}'>>"$MYTMP"/platform.c
-$c "$MYTMP"/platform.c -o "$MYTMP"/platform
+$c $CFLAGS "$MYTMP"/platform.c $LDFLAGS -o "$MYTMP"/platform
 PLATFORM=`"$MYTMP"/platform`
 rm "$MYTMP"/platform "$MYTMP"/platform.c >/dev/null 2>&1
 
@@ -21,7 +21,7 @@ echo '#include <stdio.h>'>>"$MYTMP"/isnan.c
 echo '#include <stdlib.h>'>>"$MYTMP"/isnan.c
 echo '#include <math.h>'>>"$MYTMP"/isnan.c
 echo 'main(){isnan(0.0);return 0;}'>>"$MYTMP"/isnan.c
-$c "$MYTMP"/isnan.c -o "$MYTMP"/isnan $math >/dev/null 2>&1
+$c $CFLAGS "$MYTMP"/isnan.c $LDFLAGS -o "$MYTMP"/isnan $math >/dev/null 2>&1
 if [ $? = 0 ]
 then NOISNAN=
 else NOISNAN=-DNOISNAN
@@ -39,16 +39,14 @@ else dl=-ldl
      so=y
 fi
 
-opts='-O3'
-
-$c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
+$c $CFLAGS -fPIC -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
 ar rv bin/$PLATFORM/liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
 ranlib bin/$PLATFORM/liblpsolve55.a
 
 if [ "$so" != "" ]
 then
-  $c -fpic -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I. $opts $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
-  $c -shared -Wl,-Bsymbolic -Wl,-soname,liblpsolve55.so -o bin/$PLATFORM/liblpsolve55.so `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` -lc -lm -ldl
+  $c $CFLAGS -fPIC -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I. $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
+  $c $LDFLAGS -shared -Wl,-Bsymbolic -Wl,-soname,liblpsolve55.so -o bin/$PLATFORM/liblpsolve55.so `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'` -lc -lm -ldl
 fi
 
 rm *.o 2>/dev/null
-- 
2.40.1

